### ISSUE TITLE
StageAssignmentDAO incorrect function

### ISSUE THREAD

**Issue by stefesciberras** [pkp/pkp-lib#4385]
In version 3.1.1.1, function updateobject in StageAssignmentDAO does not work as intended. there seems to be a misprint: please see below

/**
	 * Update a new StageAssignment.
	 * @param $stageAssignment StageAssignment
	 */
	function updateObject($stageAssignment) {
		$this->update(
			sprintf(
				'UPDATE stage_assignments SET
					submission_id = ?,
					user_group_id = ?,
					user_id = ?,
					date_assigned = %s,
					recommend_only = ?
				WHERE	stage_assignment_id = ?',
				$this->datetimeToDB(Core::getCurrentDate())
			),
			array(
				(int) $stageAssignment->getSubmissionId(),
				$this->nullOrInt($stageAssignment->getUserGroupId()),
				$this->nullOrInt($stageAssignment->getUserId()),
				//(int) $stageAssignment->getId(),
				$stageAssignment->getRecommendOnly()?$stageAssignment->getRecommendOnly():0,
        (int) $stageAssignment->getId(),
			)
		);
	}

**Comment by asmecher** [comment]
@stefesciberras, can you clarify where exactly you're seeing a problem?

**Comment by stefesciberras** [comment]
@asmecher of course, it was late at night, so now I see my issue, I realise i did not give mroe information.

on lines 213, 214, the parameters are not in the right place. 

213   (int) $stageAssignment->getId()
214   $stageAssignment->getRecommendOnly()?$stageAssignment->getRecommendOnly():0,

This means that 

recommend_only = ?         -> gets the stage assigmment id
WHERE	stage_assignment_id = ?',   ->  gets the recommend only

Hence the select clause does not match anything except for stage_assignment_id = 1, and hence the proper row is not updated. 

A simple swap, as I have done in my previous reply, would suffice.
